packages/strapi-plugin-upload/config/functions/bootstrap.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/**
+ * An asynchronous bootstrap function that runs before
+ * your application gets started.
+ *
+ * This gives you an opportunity to set up your data model,
+ * run jobs, or perform some special logic.
+ */
+
+const path = require('path');
+const _ = require('lodash');
+const fs = require('fs');
+
+module.exports = async cb => {
+  // set plugin store
+  const pluginStore = strapi.store({
+    environment: strapi.config.environment,
+    type: 'plugin',
+    name: 'upload'
+  });
+
+  fs.readdir(path.join(strapi.config.appPath, 'node_modules'), async (err, node_modules) => {
```

Comment:
Not really related to this line but we are going to have problem with the hooks loading https://github.com/strapi/strapi/blob/master/packages/strapi/lib/core/hooks.js#L19

Reply:
Also, I think we need to think more globally because there are many plugins that will have that kind of providers system. We are already parsing the `node_modules` directory, so we need to store in the global variable the detected dependencies (I'm not sure but I think we are already doing it).

Reply:
Yes you are write, I will rename providers `upload-strapi-s3` and `upload-strapi-local` ?